### ISSUE TITLE
[ZING] SplitBrainHandlerTest failures

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#10361]
**testClustersShouldNotMergeWhenBiggerClusterIsNotActive**
```
java.lang.AssertionError: Cluster size is not correct. Expected: 2 Actual: 3
	at org.junit.Assert.fail(Assert.java:88)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:859)
	at com.hazelcast.cluster.SplitBrainHandlerTest.testClustersShouldNotMergeWhenBiggerClusterIsNotActive(SplitBrainHandlerTest.java:606)
```

https://hazelcast-l337.ci.cloudbees.com/view/Zing/job/Hazelcast-3.x-ZingJDK7/78/testReport/junit/com.hazelcast.cluster/SplitBrainHandlerTest/testClustersShouldNotMergeWhenBiggerClusterIsNotActive/

https://hazelcast-l337.ci.cloudbees.com/view/Zing/job/Hazelcast-3.x-ZingJDK8/20/testReport/junit/com.hazelcast.cluster/SplitBrainHandlerTest/testClustersShouldNotMergeWhenBiggerClusterIsNotActive/

**testClustersShouldNotMergeWhenSmallerClusterIsNotActive**
```
java.lang.AssertionError: Cluster size is not correct. Expected: 2 Actual: 3
	at org.junit.Assert.fail(Assert.java:88)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:859)
	at com.hazelcast.cluster.SplitBrainHandlerTest.testClustersShouldNotMergeWhenSmallerClusterIsNotActive(SplitBrainHandlerTest.java:643)
```
https://hazelcast-l337.ci.cloudbees.com/view/Zing/job/Hazelcast-3.x-ZingJDK8/20/testReport/junit/com.hazelcast.cluster/SplitBrainHandlerTest/testClustersShouldNotMergeWhenSmallerClusterIsNotActive/

**Comment by hasancelik** [comment]
**testClustersShouldNotMergeWhenBiggerClusterIsNotActive**
```
java.lang.AssertionError: Cluster size is not correct. Expected: 2 Actual: 3
	at org.junit.Assert.fail(Assert.java:88)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:859)
	at com.hazelcast.cluster.SplitBrainHandlerTest.testClustersShouldNotMergeWhenBiggerClusterIsNotActive(SplitBrainHandlerTest.java:606)
```

https://hazelcast-l337.ci.cloudbees.com/view/Zing/job/Hazelcast-3.x-ZingJDK6/22/testReport/junit/com.hazelcast.cluster/SplitBrainHandlerTest/testClustersShouldNotMergeWhenBiggerClusterIsNotActive/

**Comment by metanet** [comment]
Should be fixed via #10523 and #10529
